src/libtest/lib.rs

```diff
@@ -1029,6 +1051,17 @@ where
 
     let tests_len = tests.len();
 
+    if let Ok(name) = env::var(SECONDARY_TEST_INVOKER_VAR) {
+        tests.retain(|test| test.desc.name.as_slice() == name);
+        assert_eq!(1, tests.len());
+        let test = tests.into_iter().next().unwrap();
+
+        run_test(opts, !opts.run_tests, test, RunStrategy::SpawnedSecondary, Concurrent::No);
```

Comment:
This seems somewhat odd in that it's engaging a good deal of libtest infrastructure to run just the one test, I think it could just run it directly here, right?

I'm also curious what needs to be written after the test runs to stdout/stderr because the output status should be basically all we need right?

Reply:
I moved this into `run_tests_console` to bypass some steps.

> I'm also curious what needs to be written after the test runs to stdout/stderr because the output status should be basically all we need right?

Right, but now I'm experimenting with using the exit code directly to indicate the test status. This simplifies the implementation and means we don't have to think about different failure modes.

At first, I thought we'd need something like JSON to represent variants like `TrFailedMsg`. But then I realized I could just output the message to stdout and return `TrFailed`, so using the exit code is simpler.